pkg/webhook/admission/pod/logger_injector.go

```diff
@@ -0,0 +1,76 @@
+package pod
+
+import (
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	v1 "k8s.io/api/core/v1"
+	"strings"
+)
+
+const (
+	LoggerContainerName         = "inferenceservice-logger"
+	LoggerConfigMapKeyName      = "logger"
+	LoggerContainerImage        = "gcr.io/kfserving/logger"
+	LoggerContainerImageVersion = "latest"
+	PodKnativeServiceLabel      = "serving.knative.dev/service"
+)
+
+type LoggerConfig struct {
+	Image string `json:"image"`
+}
+
+type LoggerInjector struct {
+	config *LoggerConfig
+}
+
+func (il *LoggerInjector) InjectLogger(pod *v1.Pod) error {
```

Comment:
should we do this on podspec instead of injecting in mutator? knative does not support init containers but sidecar is fine right?

Reply:
This not allowed at present.  See:
  * https://github.com/knative/serving/issues/3384
  *  https://github.com/knative/serving/issues/5822